proposals/0126-type-applications-in-patterns.rst

```diff
@@ -85,16 +130,20 @@ A more complex example is this (also inspired by `#15050 <https://gitlab.haskell
       MkT3 :: forall a b.            T a
       MkT4 :: forall a b. b ~ Int => T a
       MkT5 :: forall a b c. b ~ c => T a
+      MkT6 :: forall a b. ClassWithFunDep a b => T a
+
+    class ClassWithFunDep a b | a -> b
 
     foo :: T (Int, Int) -> ()
     foo (MkT1 @(Int,Int))  = ()
     foo (MkT2 @x)          = (() :: x ~ Int => ())
-    foo (MkT3 @_ @x)       = (() :: x ~ x => ()) -- nb: x is in scope
+    foo (MkT3 @_ @x)       = (()) -- x is unconstrained
     foo (MkT4 @_ @x)       = (() :: x ~ Int => ())
     foo (MkT4 @_ @Int)     = ()
-    foo (MkT5 @_ @x @x)    = (() :: x ~ x => ())
+    foo (MkT5 @_ @x @y)    = (() :: x ~ y => ())
+    foo (MkT6 @x @y)       = (() :: ClassWithFunDep x y => ()) -- hard to bind b otherwise!
 
-All of these equations type-check (just like they would if added value arguments of type ``a``, ``b``,... to the constructors and turned the type applications into type signatures).
+All of these equations type-check.
```

Comment:
Did you remove this sentence because itis no longer true?

Reply:
Yes, but I was wrong. Those are all fine; will revert.

Reply:
```suggestion
All of these equations type-check (just like they would if added value arguments of type ``a``, ``b``,... to the constructors and turned the type applications into type signatures).
```